include/unifex/bulk_join.hpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2019-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/tag_invoke.hpp>
+#include <unifex/execution_policy.hpp>
+#include <unifex/get_execution_policy.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+
+namespace _bulk_join {
+
+template<typename Receiver>
+struct _join_receiver {
+    class type;
+};
+
+template<typename Receiver>
+using join_receiver = typename _join_receiver<Receiver>::type;
+
+template<typename Receiver>
+class _join_receiver<Receiver>::type {
+public:
+    template<typename Receiver2>
+    explicit type(Receiver2&& r) noexcept(std::is_nothrow_constructible_v<Receiver, Receiver2>)
+    : receiver_((Receiver2&&)r)
+    {}
+
+    void set_next() & noexcept {}
```

Comment:
I'm not sure I follow how an empty body here constitutes tracking all of the `set_next` calls and completing once all of the `set_next()` calls have completed, as your spec says above. What am I missing?

Reply:
The calls to `set_next()` return before `set_done()/set_value()/set_error()` is called.
So we just discard the (empty) "next value" here and then rely on the normal sender signals forwarding through to the single-sender receiver to signal that the bulk operation has completed.